src/lib/datepicker/datepicker.ts

```diff
@@ -168,7 +168,18 @@ export class MatDatepicker<D> implements OnDestroy {
   @Output() selectedChanged = new EventEmitter<D>();
 
   /** Whether the calendar is open. */
-  opened = false;
+  @Input()
+  get opened(): boolean { return this._opened; }
+  set opened(value: boolean) {
+    const shouldOpen = coerceBooleanProperty(value);
+
+    if (shouldOpen) {
```

Comment:
You could shorten this one down with a ternary: `shouldOpen ? this.open() : this.close()`.

Reply:
Done.